pkg/util/runtimeconfig/manager.go

```diff
@@ -14,7 +15,7 @@ import (
 )
 
 // Loader loads the configuration from file.
-type Loader func(filename string) (interface{}, error)
+type Loader func(filename string) (cfg interface{}, hash []byte, error error)
```

Comment:
Hmm, changing this will break other users (Loki).

Reply:
In theory it would probably be better to do the hashing outside of the Loader.

I guess that would require us to read the config file twice, which might actually read two different versions.

I think if we really change the interface here I would suggest we would use a `func(reader io.ReadCloser) (interface{}, error)` and then we can be quite flexible for future use cases

Reply:
I think that would be fine too...

Current approach gives us a different flexibility -- namely that Loader can load config from anywhere, and can treat `filename` as URL, or even config data itself. But we don't really need that :) Let's use `func(reader io.ReadCloser) (interface{}, error)` and do hashing in Manager itself.